hibernate-core/src/test/java/org/hibernate/engine/internal/StatisticalLoggingSessionEventListenerTest.java

```diff
@@ -0,0 +1,226 @@
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.internal.CoreMessageLogger;
+
+import org.hibernate.testing.logger.LoggerInspectionRule;
+import org.hibernate.testing.logger.Triggerable;
+import org.hibernate.testing.orm.domain.gambit.BasicEntity;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.Jira;
+import org.hibernate.testing.orm.junit.ServiceRegistry;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.orm.junit.Setting;
+import org.junit.Rule;
+import org.junit.jupiter.api.Test;
+
+import org.jboss.logging.Logger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * @author Christian Bürgi
+ */
+@DomainModel(annotatedClasses = {
+		BasicEntity.class
+})
+@SessionFactory(generateStatistics = true)
+@ServiceRegistry(settings = {
+		@Setting(name = AvailableSettings.STATEMENT_BATCH_SIZE, value = "10"),
+		@Setting(name = AvailableSettings.ORDER_INSERTS, value = "true"),
+		@Setting(name = AvailableSettings.USE_QUERY_CACHE, value = "true"),
+		@Setting(name = AvailableSettings.USE_SECOND_LEVEL_CACHE, value = "true")
+})
+@Jira("https://hibernate.atlassian.net/browse/HHH-18513")
+public class StatisticalLoggingSessionEventListenerTest {
+
+	@Rule
+	public LoggerInspectionRule logInspection = new LoggerInspectionRule(
+			Logger.getMessageLogger(
+					CoreMessageLogger.class,
+					StatisticalLoggingSessionEventListener.class.getName()
+			) );
+
+	@Test
+	void testSessionMetricsLog(SessionFactoryScope scope) {
+		Triggerable triggerable = logInspection.watchForLogMessages( "Session Metrics {" );
+		long startTime = System.nanoTime();
+
+		scope.inTransaction( session -> {
+			session.persist( new BasicEntity( 1, "fooData" ) );
+			session.persist( new BasicEntity( 2, "fooData2" ) );
+			session.flush();
+			session.clear();
+			for ( int i = 0; i < 2; i++ ) {
+				assertThat( session.createQuery( "select data from BasicEntity e where e.id=1", String.class )
+									.setCacheable( true )
+									.getSingleResult() ).isEqualTo( "fooData" );
+			}
+		} );
+		long sessionNanoDuration = System.nanoTime() - startTime;
+
+		List<String> messages = triggerable.triggerMessages();
+		assertThat( messages ).hasSize( 1 );
+		String sessionMetricsLog = messages.get( 0 );
+
+		// acquiring JDBC connections
+		SessionMetric acquiringJdbcConnectionsMetric = extractMetric(
+				sessionMetricsLog,
+				"([0-9]+) nanoseconds spent acquiring ([0-9]+) JDBC connections"
+		);
+		assertThat( acquiringJdbcConnectionsMetric.getDuration() ).isGreaterThan( 0L )
+				.isLessThan( sessionNanoDuration );
+		assertThat( acquiringJdbcConnectionsMetric.getCount() ).isEqualTo( 1 );
```

Comment:
Is it ok to check for the actual count metrics or do you expect them to be too unstable? Certain changes in the internal logic could result in test failures here.

Reply:
probably better to avoid this check

Reply:
I removed the count checks